src/lifx.d.ts

```diff
@@ -317,6 +335,34 @@ export class Light {
   reboot(callback: any): void;
 }
 
+export interface Product {
+  vendorName: string;
+  productName: string;
+  productFeatures: Features;
+  productUpgrades: Upgrade[];
+}
+
+export interface Features {
+  hev?: boolean;
+  color: boolean;
+  chain: boolean;
+  matrix: boolean;
+  relays?: boolean;
+  buttons?: boolean;
+  infrared: boolean;
+  multizone: boolean;
+  temperature_range?: number[] | null;
+  extended_multizone?: boolean;
+  min_ext_mz_firmware?: number;
+  min_ext_mz_firmware_components?: number[];
+}
```

Comment:
I skimmed through the `products.json` and all but the 3 last ones seem to be always there. Temperature range is never `null`. My suggestion:

```ts
export interface ProductFeatures {
  hev: boolean;
  color: boolean;
  chain: boolean;
  matrix: boolean;
  relays: boolean;
  buttons: boolean;
  infrared: boolean;
  multizone: boolean;
  temperature_range: [number, number];
  extended_multizone?: boolean;
  min_ext_mz_firmware?: number;
  min_ext_mz_firmware_components?: number[];
}
```